src/core_codemods/combine_startswith_endswith.py

```diff
@@ -17,36 +18,54 @@ class CombineStartswithEndswith(SimpleCodemod, NameResolutionMixin):
     def leave_BooleanOperation(
         self, original_node: cst.BooleanOperation, updated_node: cst.BooleanOperation
     ) -> cst.CSTNode:
-        if not self.filter_by_path_includes_or_excludes(
-            self.node_position(original_node)
-        ):
-            return updated_node
-
         if self.matches_startswith_endswith_or_pattern(original_node):
-            left_call = cst.ensure_type(updated_node.left, cst.Call)
-            right_call = cst.ensure_type(updated_node.right, cst.Call)
+            if not self.filter_by_path_includes_or_excludes(
```

Comment:
any reason you moved this down? It makes more sense to stay where it was so no work happens if this node is excluded. Also you could combine the if statements if you like.

Reply:
You're right, this was a mistake. I originally did this with the decorator below and just copied the code into the wrong spot when refactoring to try to make minimal changes. 

However I noticed that a lot of the codemods begin with a statement to check filters by includes/excludes and check that a node is not selected so I think a decorators could be cleaner and reduce repetition for these cases.

See [utils_decorators](https://github.com/LucasFaudman/codemodder-python/blob/main/src/codemodder/codemods/utils_decorators.py) in my fork.

```python
def check_filter_by_path_includes_or_excludes(
    func: Callable[[_BaseVisitorOrTransformer, _CSTNode, _CSTNode], cst.CSTNode]
) -> Callable[[_BaseVisitorOrTransformer, _CSTNode, _CSTNode], cst.CSTNode]:
    """
    Decorator to filter nodes based on path-includes or path-excludes flags.

    Calls the decorated func only if the original node is not filtered.
    otherwise, returns the updated node as is.

    ```
    @check_filter_by_path_includes_or_excludes
    def leave_Call(self, original_node: cst.Call, updated_node: cst.Call) -> cst.Call
        # rest of the method
    ```

    Is equivalent to:

    ```
    def leave_Call(self, original_node: cst.Call, updated_node: cst.Call) -> cst.Call
        # Added by the decorator
        if not self.filter_by_path_includes_or_excludes(
            self.node_position(original_node)
            ):
            return updated_node

        # rest of the method
    ```
    """

    @wraps(func)
    def wrapper(
        instance: _BaseVisitorOrTransformer,
        original_node: _CSTNode,
        updated_node: _CSTNode,
    ) -> cst.CSTNode:
        if not instance.filter_by_path_includes_or_excludes(
            instance.node_position(original_node)
        ):
            return updated_node
        return func(instance, original_node, updated_node)

    return wrapper
```

Reply:
I like the decorator pattern! Some codemods don't need this excludes check which is why we didn't generalize like this, but there are enough that do that the pattern looks nice. @drdavella can give his opinion too